src/Component/src/Metadata/Metadata.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata;
+
+use Doctrine\Inflector\Inflector as InflectorObject;
+use Doctrine\Inflector\InflectorFactory;
+
+final class Metadata implements MetadataInterface
+{
+    private string $name;
+
+    private string $applicationName;
+
+    /** @var string */
+    private $driver;
```

Comment:
Outside of scope of current PR, but I cannot not comment :smiling_face_with_tear: 

```suggestion
    private string|false $driver;
```

Reply:
This Metadata object will be removed at some point, I've build another metadata system. So I don't think it's useful to improve it :)